core/src/main/java/io/questdb/cairo/TableWriter.java

```diff
@@ -5308,6 +5306,32 @@ private boolean dropPartitionByExactTimestamp(long timestamp) {
         return true;
     }
 
+    private void enforceTTL() {
+        int ttl = metadata.getTTL();
+        if (ttl == 0) {
+            return;
+        }
+        long oldDataCutoffMicros = getMaxTimestamp() - TimeUnit.HOURS.toMicros(ttl);
```

Comment:
when this works, it is important to optimise this code to ensure commit latency is low. For example, if we can deduce from min/max timestamps that partition traversal is not required - we should do so

Reply:
The updated algo starts with the oldest partition, and stops as soon as reaching a partition that's not evictable.

It decides whether a partition is evictable by using the `ceil` function on the partition timestamp.